### DISCUSSION TITLE
Facetracking using KinectSensor

### DISCUSSION THREAD

**Discussion by tyler-lynch** [microsoft/psi#183] [Q&A]
Hi everyone,

I am trying to figure out how to draw a bounding box around a face using the KinectFaceTracker. 

There seems to be a lot of useful properties in Microsoft.Kinect.Face or KinectFace. For example FaceBoundingBoxInColorSpace looks useful but I am not sure how to use it or any of the other properties in KinectFace and Microsoft.Kinect.Face. Is there any documentation on how to use these?

Anything is appreciated. 

Thanks!

**Comment by sandrist** [comment]
Hi,

The documentation for the Kinect component can be found [here](https://github.com/microsoft/psi/wiki/Kinect-Overview), including a very simple sample on how to use the Kinect face tracker component. These \psi components basically try to wrap much of the core Kinect SDK functionality, which is documented for example here: [FaceBoundingBoxInColorSpace Property](https://docs.microsoft.com/en-us/previous-versions/windows/kinect/dn773608(v=ieb.10))

If you're wondering how to visualize the bounding box, we don't have visualizers for the `KinectFace` or `RectI` types directly, but it's fairly easy to use a `Select` operator to convert the `KinectFace`s into `System.Drawing.Rectangle`s, which we do have default visualizers for.

For example, something like:
```csharp
...
var faceRectangles = kinectFaceDetector.Faces.Select(faceList => faceList.Select(face => new System.Drawing.Rectangle(...)).ToList()).Write(...);
```
